vendor/knative.dev/hack/library.sh

```diff
@@ -718,18 +718,23 @@ function __go_update_deps_for_module() {
     fi
   fi
 
-  group "Go mod tidy and vendor"
+  group "Go mod tidy"
 
   # Prune modules.
-  local orig_pipefail_opt=$(shopt -p -o pipefail)
+  local orig_pipefail_opt
+  orig_pipefail_opt=$(shopt -p -o pipefail)
   set -o pipefail
   go mod tidy 2>&1 | grep -v "ignoring symlink" || true
-  go mod vendor 2>&1 |  grep -v "ignoring symlink" || true
+  if [[ "${FORCE_VENDOR:-false}" == "true" ]] || [ -d vendor ]; then
+    group "Go mod vendor"
+    go mod vendor 2>&1 |  grep -v "ignoring symlink" || true
+  fi
   eval "$orig_pipefail_opt"
 
-  if ! [ -d vendor ]; then
+  if ! [[ "${FORCE_VENDOR:-false}" == "true" ]] && ! [ -d vendor ]; then
```

Comment:
@cardil rather than an env var why don't we just infer repos want to use vendor if they already have a `vendor` folder?

This would mean things should just work as expected until repos remove vendor themselves?

Reply:
Another question are we skipping `update_licenses` call if there is no vendor? If so why?

Reply:
It already works like you're suggesting.

If you don't have a vendor dir, it will not be created for you. If you have one, it will be updated as it was before. You could set this environment flag to force the creation of vendor dir, which I imagine being handy for Knative's vendors.

The license saving is removed here, yes, if there's no vendor dir. But, remember we still do the license scan during the `--build-test` run. I believe that satisfies the CNCF license guidelines AFAIK.

Reply:
> The license saving is removed here, yes, if there's no vendor dir. But, remember we still do the license scan during the --build-test run. I believe that satisfies the CNCF license guidelines AFAIK.

We ship licenses in our containers - we shouldn't stop doing that unless we get clear guidance from the CNCF that we no longer need to do that

Reply:
So I would expect we still run even if we don't have a vendor folder

`update_licenses third_party/VENDOR-LICENSE "./..."`